cmd/brew-test-bot.rb

```diff
@@ -552,18 +553,9 @@ def skip(formula_name)
     end
 
     def satisfied_requirements?(formula, spec, dependency = nil)
-      requirements = formula.send(spec).recursive_requirements
-
-      unsatisfied_requirements = requirements.reject do |requirement|
-        satisfied = false
-        satisfied ||= requirement.satisfied?
-        satisfied ||= requirement.optional?
-        if !satisfied && requirement.default_formula?
-          default = Formula[requirement.default_formula]
-          satisfied = satisfied_requirements?(default, :stable, formula.full_name)
-        end
-        satisfied
-      end
+      f = Formulary.factory(formula.full_name, spec)
+      fi = FormulaInstaller.new(f)
+      unsatisfied_requirements, _ = fi.expand_requirements
```

Comment:
I agree that it feels weird. The reason I used `FormulaInstaller` was for the `install_bottle_for?` logic, which helps determine when non-runtime requirements aren't necessary ([this logic branch specifically](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/formula_installer.rb#L446-L447)).